metrics/api/src/main/java/io/helidon/metrics/api/MeterRegistry.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics.api;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+/**
+ * Manages the look-up and registration of meters.
+ */
+public interface MeterRegistry extends Wrapped {
+
+    /**
+     * Returns all previously-registered meters.
+     *
+     * @return registered meters
+     */
+    List<? extends Meter> meters();
+
+    /**
+     * Returns previously-registered meters which match the specified {@link java.util.function.Predicate}.
+     *
+     * @param filter the predicate with which to evaluate each {@link io.helidon.metrics.api.Meter}
+     * @return meters which match the predicate
+     */
+    Collection<? extends Meter> meters(Predicate<Meter> filter);
+
+    /**
+     * Returns the default {@link io.helidon.metrics.api.Clock} in use by the registry.
+     *
+     * @return default clock
+     */
+    Clock clock();
+
+    /**
+     * Locates a previously-registered meter using the name and tags in the provided builder or, if not found, registers a new
+     * one using the provided builder.
+     *
+     * @param builder builder to use in finding or creating a meter
+     * @return the previously-registered meter with the same name and tags or, if none, the newly-registered one
+     * @param <M> type of the meter
+     * @param <B> builder for the meter
+     */
+    <M extends Meter, B extends Meter.Builder<B, M>> M getOrCreate(B builder);
+
+    /**
+     * Locates a previously-registered counter.
+     *
+     * @param name name to match
+     * @param tags tags to match
+     * @return {@link java.util.Optional} of the previously-registered counter; empty if not found
+     */
+    default Optional<Counter> getCounter(String name, Iterable<Tag> tags) {
```

Comment:
Yes, deliberate. There is some rationale to the naming.

* For individual meter types, the "get-less" method names have "get-or-register a single meter" semantics; the "getXXX" methods are pure gets. This is what both Micrometer and MicroProfile do in their respective APIs. 
* For the `meters` methods which return (potentially) multiple meters I followed our naming convention guideline in which we avoid the "get" prefix in general. 

Perhaps changing the `meters` methods to `getMeters` would be clearer. Even though it would be counter to our general guidelines, it might actually be better in this case for consistency with the "get-less" vs. "get" semantics on the single-meter methods.

Interested in feedback on this.